roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/util/ConfigParserTest.java

```diff
@@ -117,13 +121,95 @@ public void configParserTest_generateModel() throws ConfigParserException {
+    }
+
+    @Test
+    public void configParserTest_structureCorrect() throws ConfigParserException {
+        Experiment exp = new Experiment("Experiment");
+        IRoundaboutStructure roundaboutStructure = configParser.generateRoundaboutStructure(roundAboutConfig, exp);
+
+        for (IStreetConnector connector : roundaboutStructure.getStreetConnectors()) {
+            Integer exitSectionId = null;
+            Integer nextTrackSectionId = null;
+            for (IConsumer section : connector.getNextSections()) {
+                if (section.toString().contains("exit")) {
+                    exitSectionId = extractSectionId(section.toString());
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (nextTrackSectionId != null && nextTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    nextTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and exit on next connector", false);
+                }
+            }
+
+            Integer entrySectionId = null;
+            Integer previousTrackSectionId = null;
+            for (IProducer section : connector.getPreviousSections()) {
+                if (section.toString().contains("entry")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (entrySectionId != null && entrySectionId != id) {
+                        assertTrue("entries from different sections on next connector", false);
+                    }
+                    entrySectionId = id;
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (previousTrackSectionId != null && previousTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    previousTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and entries on previous connector", false);
+                }
+            }
+
+            assertTrue("only tracks from neighbour sections", areNeighbourSections(previousTrackSectionId, nextTrackSectionId, roundaboutStructure.getStreetConnectors().size()));
+            if (exitSectionId != null) { // TODO fix after connector implementation
+                assertEquals("exit and previous tracks from same section", exitSectionId, previousTrackSectionId);
+            }
+            if (entrySectionId != null) { // TODO fix after connector implementation
+                assertEquals("entries and next tracks from same section", entrySectionId, nextTrackSectionId);
+            }
+        }
+    }
```

Comment:
although this test works, did you notice the huge amount of logic in a test?
a test should not contain (almost) any logic, otherwise the test it self could have bugs
it seems like you check every item, .. again, a lot of asserts and its not visible on which item it does fail
consider using helper methods to compare arrays/lists or/and try to use fluent assertions like AssertJ to increase readability

Reply:
this one needs logic, because i'm testing the correctness of the structure object ... this test checks, if the xml is loaded correctly into the structure object ... but we can start an test refactoring branch, if you have a good idea how to get rid of that nested logic

Reply:
i would have to think about how to get rid of the nested logic, but would do it.
the thing is, that the project is almost over (1 month) and we have the same old problem, that we need to get all features fast ;D
not sure how vollbrecht will read it (and how he gives us our mark) - otherwise it seems to be wasted time